src/factories.js

```diff
@@ -30,23 +34,23 @@ const mergePropsAndClassName = (props, extraProps) => {
  * @param {function|string} Component A ReactClass or string
  * @param {function} mapValueToProps A function that maps a primitive value to the Component props
  * @param {string|object|function} val The value to create a ReactElement from
- * @param {object} extraProps Additional props to add to the final ReactElement
+ * @param {object} defaultProps Default props to add to the final ReactElement
  * @returns {function|null}
  */
-export function createShorthand(Component, mapValueToProps, val, extraProps = {}) {
+export function createShorthand(Component, mapValueToProps, val, defaultProps = {}) {
   // Clone ReactElements
   if (isValidElement(val)) {
-    return React.cloneElement(val, mergePropsAndClassName(val.props, extraProps))
+    return React.cloneElement(val, mergePropsAndClassName(defaultProps, val.props))
   }
 
   // Create ReactElements from props objects
   if (_.isPlainObject(val)) {
-    return <Component {...mergePropsAndClassName(val, extraProps)} />
+    return <Component {...mergePropsAndClassName(defaultProps, val)} />
   }
 
   // Map values to props and create a ReactElement
   if (_.isString(val) || _.isNumber(val)) {
-    return <Component {...mergePropsAndClassName(mapValueToProps(val), extraProps)} />
+    return <Component {...mergePropsAndClassName(defaultProps, mapValueToProps(val))} />
```

Comment:
The `defaultProps` will always be what we (the stardust authors) set as the default. The props, whether passed directly or derived from `mapValueToProps`, will be what the end-user wants. So those should always win.

Take this example component that has an image with a default src:
```js
// Example:
const Foo = (props) => (
  <div>
    createShorthand(Image, val => ({ src: val }), props.image, { src: 'foo.png' })
  </div>
)
```

The different usages are:

React Element:
```js
<Foo image={<MyImage src='bar.png'> />

// Within createShorthand:
// Clone ReactElements
if (isValidElement(val)) {
  return React.cloneElement(val, mergePropsAndClassName(defaultProps, val.props))
}
```

Props object
```js
<Foo image={{ src: 'bar.png' }} />

// Within createShorthand:
// Create ReactElements from props objects
if (_.isPlainObject(val)) {
  return <Component {...mergePropsAndClassName(defaultProps, val)} />
}
```


Primitive:
```js
<Foo image='bar.png' />

// Within createShorthand:
// Map values to props and create a ReactElement
if (_.isString(val) || _.isNumber(val)) {
  return <Component {...mergePropsAndClassName(defaultProps, mapValueToProps(val))} />
}
```

In all of these cases, if we swapped the argument order of `defaultProps` the src would not be what the user passed.